### PR TITLE
ceph-daemon: a few fixes; functional test

### PR DESCRIPTION
**Pull request by liewegas**
- functional tests covering everything but 'run' and 'adopt'
- 'exec' is removed/combined with 'enter'
- 'shell' (and 'enter') take optional commands, otherwise run bash

### CODE REVIEW

File: qa/standalone/test_ceph_daemon.sh [hunk 1]
```diff
@@ -0,0 +1,135 @@
+#!/bin/bash -ex
+
+[ -z "$SUDO" ] && SUDO=sudo
+[ -x ../src/ceph-daemon ] && CEPH_DAEMON=../src/ceph-daemon
+[ -x ./ceph-daemon ] && CEPH_DAEMON=.ceph-daemon
+which ceph-daemon && CEPH_DAEMON=$(which ceph-daemon)
+
+FSID='00000000-0000-0000-0000-0000deadbeef'
+IMAGE='ceph/daemon-base:latest-master'
```
Comment (sebastian-philipp) [inline qa/standalone/test_ceph_daemon.sh]: Can we make use of the new images on quay.io

File: src/ceph-daemon [hunk 2]
```diff
@@ -1189,6 +1188,8 @@ def command_ls():
             elif is_fsid(i):
                 fsid = i
                 for j in os.listdir(os.path.join(args.data_dir, i)):
+                    if '.' not in j:
```
Comment (sebastian-philipp) [inline src/ceph-daemon]: Seems dangerous to me. Can we make similar to
```suggestion
                    if not j.startswith('./'):
```
?

File: qa/standalone/test_ceph_daemon.sh (hunk 1 above)
Comment (liewegas) [reply qa/standalone/test_ceph_daemon.sh]: which image?  right now quay.io only gets the wip- branches for some annoying ceph-build reason

File: src/ceph-daemon (hunk 2 above)
Comment (liewegas) [reply src/ceph-daemon]: this is really there to skip the 'crash' directory.  but probably we should be tolerant of 'a.b.c' too. will make this more robust

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
retest this please

**Pull request merged**